testng-core/src/main/java/org/testng/TestNG.java

```diff
@@ -1403,7 +1412,12 @@ public static TestNG privateMain(String[] argv, ITestListener listener) {
    *
    * @param cla The command line parameters
    */
-  protected void configure(CommandLineArgs cla) {
+  public void configure(CommandLineArgs cla) {
```

Comment:
It is protected because it is not supposed to be used by users.
In fact, the CLI runner should call the appropriate methods on `TestNG` and `TestNG` should not have any dependency to CLI runner.

Reply:
The javadocs on the method that accepts a Map says its deprecated and favours this method. With it being protected i wouldn't be able to test it out. Are you suggesting that i create a subclass of TestNG for testing purposes wherein i make it public ?

Reply:
This method is currently what is being proposed as an alternative to surefire plugin invocation (it currently calls a method that accepts a `Map`). So am guessing that this method has to be public else surefire plugin wouldn't be able to invoke it (Unless and until they are subclassing TestNG and then invoking this. Are they ? )

Reply:
Right, good catch. Time passed.
Now, I think we should have a dedicated interface if we want to open the method to `public`.

```java
public void configure(Configuration configuration) { ... }
class CommandLineArgs implements Configuration { ... }
```

Reply:
I dont think that would work.. I mean `implements Configuration` part because I dont know if the configuration captures everything that TestNG needs. Setting up TestNG via configurations is right now spread all over the place. In some places, TestNG sets it directly and in some cases it  delegates to the Configuration object. I think this would be an exercise by itself.

So in the meantime, do you think that I should just subclass TestNG for the tests wherein I set this configuration and complete this PR ?

Reply:
Or maybe having a `TestNGConfigurator`:

```java
interface TestNGConfigurator {
  void configure(TestNG tng);
}
```

Reply:
I didnt understand on 
* Who would implement this ?
* How would this be consumed ?
* Is this meant for tests only ?

Reply:
> So in the meantime, do you think that I should just subclass TestNG for the tests wherein I set this configuration and complete this PR ?

It looks better in short term.

> I didnt understand on
>
> Who would implement this ?
> How would this be consumed ?
> Is this meant for tests only ?

It is a global thinking. I see a bad dependency between CLI runner and core module, and try to find a way to move the dependency from `core -> cli` to `cli -> core`.

Reply:
Ok.. then I will for now refactor the PR to sub class TestNG and be done with this PR.

Reply:
Thanks. If it takes too much effort, check if another compromise is possible without opening the method.

Reply:
sub-classed TestNG and reverted back the public method to its `protected` state.